test/export/test_export.py

```diff
@@ -1672,6 +1672,94 @@ def forward(self, x):
+    def test_subclass_nested_attr_access(self):
+        from torch.testing._internal.two_tensor import TwoTensor
+
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.p1 = torch.nn.Parameter(torch.ones(3, 4))
+                self.p2 = torch.nn.Parameter(
+                    TwoTensor(
+                        TwoTensor(torch.ones(3, 4), torch.ones(3, 4)),
+                        TwoTensor(torch.ones(3, 4), torch.ones(3, 4)),
+                    )
+                )
+
+            def forward(self, x):
+                res = (2 * self.p1 + self.p2).sum()
+                return x + res.get_elem_a().b
+
+        m = Foo()
+        ref_x = torch.randn(3, 4)
+        ref_out = m(ref_x)
+        ep_training = torch.export.export_for_training(m, (ref_x,), strict=False)
+        self.assertExpectedInline(
+            str(ep_training.graph).strip(),
+            """\
+graph():
+    %p_p1 : [num_users=1] = placeholder[target=p_p1]
+    %p_p2 : [num_users=1] = placeholder[target=p_p2]
+    %x : [num_users=1] = placeholder[target=x]
+    %mul : [num_users=1] = call_function[target=torch.ops.aten.mul.Tensor](args = (%p_p1, 2), kwargs = {})
+    %add : [num_users=1] = call_function[target=torch.ops.aten.add.Tensor](args = (%mul, %p_p2), kwargs = {})
+    %sum_1 : [num_users=1] = call_function[target=torch.ops.aten.sum.default](args = (%add,), kwargs = {})
+    %getattr_33 : [num_users=1] = call_function[target=builtins.getattr](args = (%sum_1, a), kwargs = {})
+    %getattr_38 : [num_users=1] = call_function[target=builtins.getattr](args = (%getattr_33, b), kwargs = {})
+    %add_1 : [num_users=1] = call_function[target=torch.ops.aten.add.Tensor](args = (%x, %getattr_38), kwargs = {})
+    return (add_1,)""",
+        )
+        ep = export(m, (ref_x,))
+        self.assertTrue(torch.allclose(ep.module()(ref_x), ref_out))
+
+    @testing.expectedFailureLegacyExportNonStrict  # Old export doesn't work with subclasses
+    @testing.expectedFailureLegacyExportStrict  # Old export doesn't work with subclasses
+    @testing.expectedFailureSerDerNonStrict  # builtins.getattr is not supported  T211130564
+    @testing.expectedFailureSerDer  # builtins.getattr is not supported  T211130564
+    def test_subclass_nested_attr_access_const_metadata(self):
+        from torch.testing._internal.custom_tensor import ConstantExtraMetadataTensor
+
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.p1 = torch.nn.Parameter(torch.ones(3, 4))
+                self.p2 = torch.nn.Parameter(
+                    ConstantExtraMetadataTensor(
+                        ConstantExtraMetadataTensor(torch.ones(3, 4)),
+                    )
+                )
+
+            def forward(self, x):
+                res = 2 * self.p1 + self.p2
+                res2 = res + res.constant_attribute
```

Comment:
One thing to call out is that if you look at the expecttest result below, there isn't a node for the `res.constant_attribute` access, e.g.:
```
%constant_attribute = call_function[target=builtins.getattr](args = (res, "constant_attribute"), kwargs = {})
%add_1 = call_function[target=torch.ops.aten.add.Tensor](args = (%add, %constant_attribute), kwargs = {})
```

Instead, the constant gets inlined in the graph to its value (4):
```
%add_1  = call_function[target=torch.ops.aten.add.Tensor](args = (%add, 4), kwargs = {})
```

This... may or may not be ok. It seems fine for this example, where the constant is just an integer. Once we get constructor support working, though, then you'll have to make sure code like this can be traced properly:
```
def f(subclass):
    constant_metadata_namedtuple = subclass.metadata
    new_subclass = Subclass(subclass.elem, constant_metadata_namedtuple)
```

Where if the constant metadata is something a bit more complicated (e.g. a named tuple holding constant values, which is the case for float8), then we'll need to make sure it shows up in a reasonable way as a contructor argument.